File: RPedretti.Blazor.BingMaps/content/js/BingMaps/_Modules/_bing-maps-traffic.module.js
window.rpedrettiBlazorComponents = window.rpedrettiBlazorComponents || {};
window.rpedrettiBlazorComponents.bingMaps = window.rpedrettiBlazorComponents.bingMaps || {};
window.rpedrettiBlazorComponents.bingMaps.modules = window.rpedrettiBlazorComponents.bingMaps.modules || {};
window.rpedrettiBlazorComponents.bingMaps.modules.traffic = (function () {
    const _managers = new Map();
    return {
        init: function () {
            Microsoft.Maps.loadModule('Microsoft.Maps.Traffic', () => {
                //Create an instance of the directions manager.
                const trafficManager = new Microsoft.Maps.Traffic.TrafficManager(this.map);
                if (this.showTraffic) {
                    trafficManager.show();
                }
                _managers.set(this.mapId, trafficManager);
            });
        },
        updateTraffic: function (mapId, options) {
            _managers.get(mapId).setOptions(options);
        }
    };
})();

File: RPedretti.Blazor.BingMaps/Modules/Traffic/BingMapsTrafficModule.cs
using Microsoft.JSInterop;
using System.Threading.Tasks;

namespace RPedretti.Blazor.BingMaps.Modules.Traffic
{
    public class BingMapsTrafficModule : BaseBingMapModule
    {
        #region Fields

        private const string ModuleId = "Microsoft.Maps.Traffic";
        private string _mapId;

        #endregion Fields

        #region Properties

        private string InitFunctionName =>
            "rpedrettiBlazorComponents.bingMaps.modules.traffic.init";

        #endregion Properties

        #region Methods

        public override async Task InitAsync(string mapId)
        {
            _mapId = mapId;
            await InitModuleAsync(mapId, ModuleId, InitFunctionName, new { mapId, ShowTraffic = true });
        }

        public async Task UpateTrafficAsync(BingMapsTrafficOptions options)
        {
            await JSRuntime.Current.InvokeAsync<object>(
                "rpedrettiBlazorComponents.bingMaps.modules.traffic.updateTraffic",
                _mapId, options);
        }

        #endregion Methods
    }
}
